server/src/main/java/io/crate/execution/engine/indexing/ShardDMLExecutor.java

```diff
@@ -220,10 +223,7 @@ public CompletableFuture<TResult> apply(BatchIterator<Row> batchIterator) {
             null,
             req -> nodeLimits.get(resolveNodeId(req)).getLastRtt(TimeUnit.MILLISECONDS)
         ).consumeIteratorAndExecute()
-            .thenApply(collector.finisher())
-            .whenComplete((result, err) -> {
```

Comment:
We don't this now?

Reply:
It's because the parent accounting is closed elsewhere, right?

Reply:
Yes, see commit message:

> The parent RamAccounting instance is either closed with a task or when
the final RowConsumer finishes.